### PR TITLE
Portainer

### PR DESCRIPTION
**Pull request by alazare619**
Portainer is an advanced docker cluster,swarm, and single node management framework. Its very advanced and extremely lightweight. It integrates perfectly into rockstor i've been using it for a few weeks now without any issue managing the advanced tasks for my rockstor docker node.

Note: This is a new commit as I've changed my repo to have multiple branches as I have multiple PR's open now. Previous PR no one was reviewing or commenting for months...

### CODE REVIEW

**Comment by phillxnet** [comment]
"Previous PR no one was reviewing or commenting for months..." Yes bit of a shame that although it did get a . Hopefully this pr will receive a little more attention. The docker area is not one I'm particularly strong in so best if it's reviewed by someone that is really.
@schakrava is merge master so hopefully someone can step up and give this a go. @magicalyak may be a candidate as he's successfully review and authored a bunch of rock-on. Also it looks like this submission, like the last, has their suggestions incorporated.

I'm a bit concerned on the Rock-on intergration side given the following from your previous submission when replying to @FroggyFlox quoted question:

"@alazare619, thanks for writing this!
How does the volume creation/management function on top of Rockstor's shares? I've been tempted and a little interested by Portainer a few weeks ago but wasn't sure how to deal with this part... Would it better to create volume from within Portainer or simply bind to host's share previously created from Rockstor? I would tend to favor the latter in order to keep snapshot possibility, etc..., but I was curious about your experience and use with this so far."

"@FroggyFlox to answer your question about volumes. I've found it easier to start the dockers by hand via the cli it can use rockstors shares there. Management after the initial creation I do via portainer. I've for some time outgrew the functionality of rockstors management for docker and this definetly supplements it quiet a bit. To clarify I create the shares via rockstor. Then do the initial creation of the docker container via the cli. After that I manage all aspects of the docker via portainer."

Which if still true would warrant a little info as to it's use caveats such as we have already with for example the zoneminder and OpenVPN rockons: ie a "more_info" section that details additional requirements for their config / use. Sorry if I've missed something here, again I've never used a docker management system such as this. Just a though.

**Comment by magicalyak** [comment]
I actually did take a look at portainer, the issue is the permissions needed and also a few other items.  We may need to add a CAUTION: USE AT YOUR OWN RISK on this but I can try to add some comments below.  So sorry about the delay, had life events the last 2 months take up all my free time.

**Comment by alazare619** [comment]
Perfectly fine. I just closed it as I had a few pending PR branches I needed to submit LOL.

**Comment by alazare619** [comment]
Really to understand this is portainer is just a low level docker interface..it takes the cli commands and gives you a functioning webui...anything you can do with the cli docker can be done w/ portainer. docker attach <container name>, create a new docker network,  cleanup unused images etc..all done via webui this functionality currently doesnt exist in rockstor and while it would be nice I find it pedantic to re-invent the wheel while a very functional product already exists with a great community (Portainer)

**Comment by magicalyak** [comment]
I'd add that is really why there is a rebuilt one on Rockstor is for the btrfs and the database for the shares.  again sorry for the delay, I'll try to get through these soon.

**Comment by phillxnet** [comment]
@alazare619 Thanks for adding the warning as thoughtfully requested by @magicalyak.
I am now able to merge pull requests in this repo and would like to tend to this pr if the consensus is that it is still viable / servers a need.
@magicalyak and @FroggyFlox I'm afraid I am going to have to call on you for your greater knowledge in this area. Can we finally move on this pr: ie does it pass review?

**Comment by FroggyFlox** [comment]
Hi @phillxnet, 
I believe your earlier summary still applies: 
 

> Which if still true would warrant a little info as to it's use caveats such as we have already with for example the zoneminder and OpenVPN rockons: ie a "more_info" section that details additional requirements for their config / use. 

@magicalyak had a similar comment:

> the issue is the permissions needed and also a few other items. We may need to add a CAUTION: USE AT YOUR OWN RISK

I fully second both of these comments and believe this needs to be included before merging. The Rock-on is definitely very useful (I use it myself) but we need to warn users of the implications it has.

The Rock-on still installs fine and can be toggled ON/OFF without problem on 3.9.1-16.

I'll try to submit a Github review soon with maybe some edits suggestions.

File: portainer.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "Portainer": {
+    "website": "https://hub.docker.com/r/portainer/portainer/",
+    "version": "1.0",
+    "description": "Advanced Users Only!!!! Portainer is a lightweight advanced management UI which allows you to easily manage your Docker host or Swarm cluster.",
```
Comment (FroggyFlox) [inline portainer.json]: As discussed in the related conversation, we can add more details and warnings on the caveats and important points to consider when using Portainer within Rockstor.

```
    "description": "<strong>Advanced Users Only!</strong> Use at your own risk.<p>Portainer is a lightweight advanced management UI which allows you to easily manage your Docker host or Swarm cluster.</p> <p>Based on the official Portainer docker image: <a href='https://hub.docker.com/r/portainer/portainer/' target='blank'>https://hub.docker.com/r/portainer/portainer/</a>",
    "more_info": "<h4>Documentation</h4><p>Please refer to Portainer's documentation for details on its functions: <a href='https://portainer.readthedocs.io/en/stable/' target='_blank'>https://portainer.readthedocs.io/en/stable/</a><h4>Important points for proper function with Rockstor</h4><p>Portainer includes a lot of features and brings new possiblities to manage your docker environment. Nevertheless, please note that it is running alonside Rockstor, meaning that some conflicts might arise. For instance, it is important to note that all volumes automatically created and managed by Portainer will reside inside the Rocstor share designed as the Rock-ons service's root. It is thus recommended to create shares from Rockstor's webUI first, and then mount them as volume when creating a new container from within Portainer.</p>",
```

**Comment by FroggyFlox** [comment]
I just realized Github doesn't seem to provide notifications of updates on reviews, so I'm pinging @alazare619, @magicalyak, and @phillxnet here to notify them of the review above.

Cheers!

File: portainer.json (hunk 1 above)
Comment (magicalyak) [reply portainer.json]: @phillxnet maybe a feature request for an Experimental or Beta flag would be good, I think a warning should suffice here though.

**Comment by Hooverdan96** [comment]
While looking at this PR, it seems that the original image (portainer/portainer) has been deprecated. The new official one is now **portainer/portainer-ce**. It is available for amd64, arm32 and arm64, however the tag '**alpine**' has to be used ('latest' will pull the windows/amd64 version, as opposed to the alpine which was designed to be run on a linux host from what I understand).

**Comment by FroggyFlox** [comment]
Thanks a lot, @Hooverdan96 !
Agreed on all points. It seems we also need to set port 8000 in addition to 9000.
We might also want to double the requirement of the `--privileged` flag here as we shouldn't have SELinux(AppArmor) enabled on our default installs anymore. This could be problematic for those who use it, though, so my point is debatable.

**Comment by Hooverdan96** [comment]
@FroggyFlox - I think you are correct on the `--privileged` flag (when reading the documentation here: https://portainer.readthedocs.io/en/master/deployment.html.
One could add into the description (or more info) that if a user is using SELinux/AppArmor, then they might have to set up their own version with the privileged flag included. But for the majority of users that would be using this (if they do) I would assume, they'd be running Rockstor with the default settings, and therefore this flag would pose more of a risk for them?
Or, as mentioned above, since Portainer sort of supplants the Rockon framework, the information that this is a Rockon running at privileged level and is only for advanced users or something like that.
I would tend to lean to the "simpler" one of not having the privileged flag if it is not necessary in the new OpenSuse.

**Comment by phillxnet** [comment]
Closing as no response to a requested change for a long time. Please re-open if there is still interest.

**Pull request closed without merging**